kolibri/core/content/utils/annotation.py

```diff
@@ -478,6 +478,8 @@ def calculate_next_order(channel, model=ChannelMetadata):
     latest_order = model.objects.latest("order").order
     if latest_order is None:
         channel.order = 1
-    else:
+
+    if channel.order is None or channel.order == 0:
```

Comment:
I think I understand the first part of this guard (order is None), but what case does `channel.order == 0` account for?

Reply:
I found in some cases, specifically with some tests*, that channel order is actually set to 0 instead of None. I thought this would be safer since I wasn't 100% sure if channel order was set to 0 in other locations as well.

_*In `test_data_migrations.py/ChannelOrderTestCase/test_channel_order`, the channels returned from `ChannelMetadata.objects` all have an order of 0._

Reply:
Based on the model impelementation [here](https://github.com/learningequality/kolibri/blob/a1e53e541bfd8f0b4c55ef2d8faf2222039a6ab6/kolibri/core/content/models.py#L280), the default for the order field is 0, so that explains that. And the tests seem to assume that the order is strictly positive (>0).

So maybe the check for `order == None` might not be applicable?

Reply:
That makes sense!

I did find that when importing an entirely new channel, the channel used in `annotation.py` (`set_channel_metadata_fields()` and subsequently `calculate_next_order()`) did return an order of None - so I think the order field is being erased somewhere during this process, like you said.